### ISSUE TITLE
bosh deploy only shows global addon changes, not manifest/local changes

### ISSUE THREAD

**Issue by drnic** [cloudfoundry/bosh-cli#360]
I was enabling BOSH DNS with the https://github.com/cloudfoundry/bosh-deployment/blob/master/runtime-configs/dns.yml global runtime addon. I also added a manifest-specific bosh-dns-aliases job. But `bosh deploy` only displayed the proposed change to my deployment from the global addon:

```
+ addons:
+ - include:
+     stemcells:
+     - os: ubuntu-trusty
+   jobs:
+   - name: bosh-dns
+     properties:
+       cache:
+         enabled: "<redacted>"
+       health:
+         client:
+           tls: "<redacted>"
+         enabled: "<redacted>"
+         server:
+           tls: "<redacted>"
+     release: bosh-dns
+   name: bosh-dns
```

The output was missing the changes/additions of a manifest specific addon. The addon itself was applied; only the output above was wrong.

**Comment by dpb587-pivotal** [comment]
:link: [Story #152909631](https://www.pivotaltracker.com/story/show/152909631) &mdash; Deployment-level addons fully show up in diff

**Comment by keymon** [comment]
+1

**Comment by voelzmo** [comment]
Encountered the same issue, which had me stop and wonder for a while. It seems that global `addons` from the runtime config override local `addons` from the manifest in the diff.

Steps to reproduce:
* `bosh deploy` something that uses an `addons` section in the manifest
* `bosh urc` a runtime config that contains an `addons` section
* `bosh deploy` again and look at the diff

Expected:
* both `addons` sections and their necessary `releases` counterparts are merged together

Actual:
* Only `addons` and `releases` from the global definitions in the runtime config show up in the diff as "added"
* The `addons` and `releases` from the manifest local defintions show up in the diff as "removed"

Note:
* This seems to be related to the diff only, deployment is a no-op. The addons defines in the manifest are not removed from the instances

Bonus bug:
* This behavior even occurs, if the global `addons` section in the runtime config *excludes* the specific deployment. Expected behavior here would be: no changes in the diff at all, as nothing is applied to the deployment

**Comment by ansd** [comment]
The same bug applies to `tags`.

**Comment by scottschulthess** [comment]
As bosh dns rolls out this is very confusing, (that and https://github.com/cloudfoundry/bosh/issues/1862)

**Comment by voelzmo** [comment]
This should be fixed with bosh [v268.4.0](https://github.com/cloudfoundry/bosh/releases/tag/v268.4.0). Please re-open if you keep seeing this.
